### DISCUSSION TITLE
Rails scaffold generated RSpec doesn't pass Rubocop

### DISCUSSION THREAD

**Discussion by pedz** [rubocop-hq/rubocop-rspec#1404] [Q&A]
The test that was generated is:
```ruby
require 'rails_helper'

RSpec.describe 'import_files/new', type: :view do
  before do
    assign(:import_file, build(:import_file))
  end

  it 'renders new import_file form' do
    render

    assert_select 'form[action=?][method=?]', import_files_path, 'post' do
      assert_select 'input[name=?]', 'import_file[path]'
      assert_select 'select[name=?]', 'import_file[file_type]'
    end
  end
end
```
Rubocop complains with "RSpec/NoExpectationExample: No expectation found in this example."

I tracked down [RSpec/NoExpectationExample](https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecnoexpectationexample) and it has something I can add to the `.rubocop.yml` file.  I tried:
```yaml
RSpec:
  Language:
    Expectations:
      - assert_select
```
but now I get "Example has too many expectations [3/1]".  Which is true but the outer assert_select is doing two things.  It is making sure the form exists but it is also grabbing its children and passing it down to the two inner selects.  So... I don't see how to write this using just one assert_select.

And curiously, [this example](https://rspec.rubystyle.guide/#view-assign) from the RSpec style guide is doing pretty much the same thing.  The outer expect is testing things and then passing things down to the inner expect.

So... my guess is that somehow I can tell Rubocop via `.rubocop.yml` that assert_select works the same way.  But... looking at [default.yml](https://github.com/rubocop/rubocop-rspec/blob/master/config/default.yml) I don't see `expect` anywhere except under `Expectations` so I'm stumped.

**Comment by pirj** [comment]
The RSpec style guide [mildly suggests](https://rspec.rubystyle.guide/#expectation-per-example) that one expectation per example [is legacy](https://github.com/rubocop/rspec-style-guide/pull/47).

Would you like to harvest [some real-world usages](https://github.com/pirj/real-world-rspec) and see if this is mostly ignored, or does the one expectation per example still stand?

I always tend to tweak the `RSpec/MultipleExpectations`'s `Max` setting.

**Comment by pedz** [comment]
I'm very new to all this.  The real world usages I think will help me a lot.  I've been hoping to discover some project that I can refer to for examples, depth of testing, etc.

The `:aggregate_failures` I've noticed while surfing around but haven't looked into how to use it.  I'll do that.

In the example [the example](https://rspec.rubystyle.guide/#view-assign) I cited, would that be the right place to add `:aggregate_failures`?

But, the quest did lead me to some interesting discoveries / growth.  Although after I posted my [solution](https://stackoverflow.com/a/73980831/341980), I learned that there are even better ways to do it but I haven't spent the time yet to fully realize them.

**Reply by pirj** [reply]
It's up to you if you want to use [aggregate failures](https://relishapp.com/rspec/rspec-expectations/v/3-9/docs/aggregating-failures) or not, it would only affect how you see failure messages, just the first one or all aggregated at once.
In examples with just one expectation like the `:aggregate_failures` is useless.

For `assert_` matchers it's marginally useful, as it's from the Rails-native minitest, and may not have aggregation support in RSpec.
